lotman/lotman.go

```diff
@@ -0,0 +1,655 @@
+	if ret != 0 {
+		trimBuf(&errMsg)
+		return nil, errors.Errorf("Failed to determine %s's parents: %s", lotName, string(errMsg))
+	}
+
+	parents := cArrToGoArr(&cParents)
+
+	owners := []string{}
+	internalOwners := []string{}
+	for _, parent := range parents {
+		cOwners := unsafe.Pointer(nil)
+		LotmanGetLotOwners(parent, true, &cOwners, &errMsg)
+		if ret != 0 {
+			trimBuf(&errMsg)
+			return nil, errors.Errorf("Failed to determine appropriate owners of %s's parents: %s", lotName, string(errMsg))
+		}
+
+		internalOwners = append(internalOwners, cArrToGoArr(&cOwners)...)
+	}
+
+	// Deduplicate the owners
+	occurred := map[string]bool{}
+	for e := range internalOwners {
+		if !occurred[internalOwners[e]] {
+			occurred[internalOwners[e]] = true
+			owners = append(owners, internalOwners[e])
+		}
+	}
+
+	return &owners, nil
+}
+
+// Initialize the LotMan library and bind its functions to the global vars
+// We also perform a bit of extra setup such as setting the lotman db location
+func InitLotman() bool {
+	// If we've already tried to init the library, return the result of that attempt
+	if lotmanInitTried {
+		return lotmanInitSuccess
+	}
+	log.Infof("Initializing LotMan...")
+	lotmanInitTried = true
+
+	// dlopen the LotMan library
+	lotmanLib, err := purego.Dlopen(getLotmanLib(), purego.RTLD_NOW|purego.RTLD_GLOBAL)
+	if err != nil {
+		log.Errorf("Error opening LotMan library: %v", err)
+		return false
+	}
+
+	// Register LotMan funcs
+	purego.RegisterLibFunc(&LotmanVersion, lotmanLib, "lotman_version")
+	// C
+	purego.RegisterLibFunc(&LotmanAddLot, lotmanLib, "lotman_add_lot")
+	// R
+	purego.RegisterLibFunc(&LotmanGetLotJSON, lotmanLib, "lotman_get_lot_as_json")
+	// U
+	purego.RegisterLibFunc(&LotmanUpdateLot, lotmanLib, "lotman_update_lot")
+	// D
+	purego.RegisterLibFunc(&LotmanDeleteLotsRecursive, lotmanLib, "lotman_remove_lots_recursive")
+
+	// Auxilliary functions
+	purego.RegisterLibFunc(&LotmanLotExists, lotmanLib, "lotman_lot_exists")
+	purego.RegisterLibFunc(&LotmanSetContextStr, lotmanLib, "lotman_set_context_str")
+	purego.RegisterLibFunc(&LotmanGetLotOwners, lotmanLib, "lotman_get_owners")
+	purego.RegisterLibFunc(&LotmanGetLotParents, lotmanLib, "lotman_get_parent_names")
+	purego.RegisterLibFunc(&LotmanGetLotsFromDir, lotmanLib, "lotman_get_lots_from_dir")
+
+	// Set the lot_home context -- where the db lives
+	lotHome := param.Lotman_DbLocation.GetString()
+
+	errMsg := make([]byte, 2048)
+
+	log.Infof("Setting lot_home context to %s", lotHome)
+	ret := LotmanSetContextStr("lot_home", lotHome, &errMsg)
+	if ret != 0 {
+		trimBuf(&errMsg)
+		log.Errorf("Error setting lot_home context: %s", string(errMsg))
+		return false
+	}
+
+	defaultInitialized := false
+	rootInitialized := false
+
+	err = param.Lotman_Lots.Unmarshal(&initializedLots)
+	if err != nil {
+		log.Warningf("Error while unmarshaling Lots from config: %v", err)
+	}
+
+	federationIssuer := param.Federation_DiscoveryUrl.GetString()
```

Comment:
I think this federation Issuer fall back to the DirectorUrl may be something that's useful enough to be a utility function.